src/lib.rs

```diff
@@ -1,4 +1,5 @@
 #![no_std]
+#![feature(const_if_match)]
```

Comment:
oops, that's not even being used. I brought it in to do Vec's thing of starting with an allocation of a sensible initial capacity, but realized it doesn't really translate directly to us, because the allocation has to include the header, and I'm not sure how to calculate that. (I'll think about it)